scipio/src/sys/uring.rs

```diff
@@ -912,8 +893,8 @@ impl Reactor {
             // We are about to go to sleep. It's ok to sleep, but if there
             // is a timer set, we need to make sure we wake up to handle it.
             if let Some(dur) = timer_expiration {
-                let mut src = self.timeout_src.borrow_mut();
-                lat_ring.cancel_preempt_timer(&mut src);
```

Comment:
I think the only comment that I have is that it would be nice to force flush_cancellations to be called when we drop the timer so we don't forget to pair them. I guess it is not, because you need a wakers pointer which is different all the time, plus a reference to the ring.

Reply:
Yeah, I considered extracting this into a `self.cancel_timeout` function, but decided against it -- we only need twice, and both usages fit into the single screen. Addition a function call abstraction here would make it more difficult to understand what actually is going on here, although, tbh, I don't understand what's going on in details here anyway, especially why we need two `Option<Duration>` arguments

Reply:
@sitano should be able to explain to you, since I explained to him in his draft the other day in depth =) 
https://github.com/DataDog/scipio/pull/106

In summary one of them is the preempt timer. The other represents user timers.
We need to deal with them differently so I figured it was best to have two parameters rather than overloading one.
(They did evolve from a single parameter which is why the naming could be better, I think)

In particular, we must never sleep with the preempt timer armed. But we always sleep with the user timer (if there is any) armed.

I'll add comments and change the names a bit. This code evolved organically and could use some love by me too

Reply:
yes, that was clear to me.

Reply:
@matklad https://github.com/DataDog/scipio/pull/139